subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -318,6 +318,11 @@ static int on_publish_evt(struct mqtt_client *const client,
 	bool doc_update_rejected =
 		aws_jobs_cmp(update_topic, topic, topic_len, "rejected");
 
+#if defined(CONFIG_AWS_FOTA_LOG_LEVEL_DBG)
+	/* Fix for log print out when debugging */
+	u8_t *topic_in = (u8_t *)topic;
```

Comment:
We really should avoid this IMHO. Could you instead create a static buffer and copy the string? In that case you wouldn't need to call strdup. Or just copy it to a stack buffer before modifying. Or remove const from the topic parameter.

Reply:
What about a local stack buffer? or do we want a debug specific static buffer I'll try it out. My first iteration had a local stack buffer. The log_strdup does nothing if you are not using deferred logging I don't see it as bad practice using it?

The only reason I did this cast was to minimize the code change while keeping the original intent but I do see it has some unfortunate side effects as you modify the receive buffer when log debug is enabled so I'll do something aligned with what you said 👍